design/template.md

```diff
@@ -0,0 +1,211 @@
+bogged down.
+-->
+
+#### Story 1
+
+#### Story 2
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+-->
+
+## Design Details
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+
+### Test Plan
+
+<!---
+Describe how the new functionality will be tested (unit tests, implementation tests, e2e tests)
+-->
+
+
+
+### Graduation Criteria
+
+<!--
+
+Describe whether the proposed functionality will be feature gated and why.
+
+Define graduation milestones and criteria if the proposed functionality should be feature-gated.
+
+Feature maturity is defined using stages alpha, beta, GA. Feature-gated functionality starts off at alpha and graduates through stages following a defined graduation criteria. A feature that is in alpha and beta stages must be off by default.
+
+References in Kubernetes documentation:
+
+[feature gate]: https://git.k8s.io/community/contributors/devel/sig-architecture/feature-gates.md
+[maturity-levels]: https://git.k8s.io/community/contributors/devel/sig-architecture/api_changes.md#alpha-beta-and-stable-versions
+
+
+Example graduation criteria:
+
+Alpha:
+
+- Feature implemented behind a feature flag
+- It is clearly defined which Kubernetes versions this feature supports
+- e2e tests pass on supported Kubernetes versions
+
+Beta:
+
+- Gather user feedback
+
+GA:
+
+- N examples of real-world usage
+- N installs
+- Allowing time for feedback
+- Works on all versions of Kubernetes supported by the version of cert-manager at which this feature becomes GA
+
+-->
+
+### Upgrade / Downgrade Strategy
+
+<!--
+Will this feature affect upgrade/downgrade of cert-manager?
+-->
+
+
+###### How can this feature be enabled / disabled for an existing cert-manager installation?
+
+<!--
+
+Consider whether any additional steps will need to be taken to start/stop using this feature.
+
+Can it be disabled after having been enabled?
+
+Do the test cases cover both the feature being enabled and it being disabled (where relevant)?
+
+-->
+
+###### Does this feature depend on any specific services running in the cluster?
+
+<!--
+For example, are external dependencies such as ingress controllers, third party CRDs etc required for this feature to function?
+-->
+
+###### Will enabling / using this feature result in any new API calls (to Kubernetes or external services)?
```

Comment:
This will be good to consider as it will include extra calls to external services like LetsEncrypt